src/main/java/org/shanerx/tradeshop/itrade/IShopCreateEventListener.java

```diff
@@ -54,14 +53,10 @@ public void onSignChange(SignChangeEvent event) {
         Block chest = findShopChest(s.getBlock());
 
         if (!player.hasPermission(getCreateIPerm())) {
-            s.setLine(0, "");
-            s.update();
-            s.setLine(1, "");
-            s.update();
-            s.setLine(2, "");
-            s.update();
-            s.setLine(3, "");
-            s.update();
+            event.setLine(0, "");
+            event.setLine(1, "");
+            event.setLine(2, "");
+            event.setLine(3, "");
```

Comment:
According to the Spigot docs, Sign#update() should *always* be called. Did it get updated without it?

Reply:
event.setLine updates on its own where s.setline and s.update seem to have problems with actually updating the line